### DISCUSSION TITLE
[FEATURE] Indication for PR is currently being reviewed by someone (specially codeowners)

### DISCUSSION THREAD

**Discussion by pranavkawle** [community/community#5542] [Pull Requests]
I work in team of 10 members who are code-owners for number of repos (GitHub Enterprise). Right now we have setup rule that we need at least 2 approvals before merging PR.

Issue is, while one of the member is in process of reviewing (started adding comments but before posting comments with approval/requesting changes), the PR gets merged. That is happening lot of time lately because of work from home condition and distributed team (by location and time-zone) environment. The one who are merging are not necessarily doing wrong, since they have 2 or more approvals already. But one who is currently reviewing may have different perspective and may have some insightful review comment.

So, my proposal is to have some kind of indication that the PR is taken under review by someone (specially codeowners). This should not block PR from getting merged (although we can add rule around this). One way to achieve this is to flag the PR when someone clicks 'Start Review' button and turn colour of merge button to Orange. When merge button is clicked, modal with info message comes up about someone currently reviewing PR.

**Comment by joseph-of-wheel** [comment]
I want this and I want it to prevent merge. I'd like to deliver my feedback all at once, but I can't, or by the time I'm done it's merged already.

I see ZERO benefit to ever actually "starting a review" since my review is meaningless until I finish.

At my last job we used a better git host that had this functionality, and I gained a reputation at my workplace for finding bugs before they shipped. Now I hardly ever bother to look at PR's because I'll never have time to do the review I feel is appropriate.

In my admittedly not-humble opinion, this should have been in the first MVP of any code review functionality at all.
